cmd/query/app/querysvc/adjuster/clockskew.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2019 The Jaeger Authors.
+// Copyright (c) 2017 Uber Technologies, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package adjuster
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/jaegertracing/jaeger/internal/jptrace"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	warningDuplicateSpanID       = "duplicate span IDs; skipping clock skew adjustment"
+	warningFormatInvalidParentID = "invalid parent span IDs=%s; skipping clock skew adjustment"
+	warningMaxDeltaExceeded      = "max clock skew adjustment delta of %v exceeded; not applying calculated delta of %v"
+	warningSkewAdjustDisabled    = "clock skew adjustment disabled; not applying calculated delta of %v"
+)
+
+// ClockSkew returns an Adjuster that corrects span timestamps for clock skew.
+//
+// This adjuster modifies the start and log timestamps of child spans that are
+// inconsistent with their parent spans due to clock differences between hosts.
+// It assumes all spans have unique IDs and should be used after SpanIDUniquifier.
```

Comment:
We should add how we decide if two spans belong to the same source/node/service. The assumption is that if they do we do not expect their timestamps to be skewed.